### PR TITLE
feat(cli): enable HypFiatCollateral warp route deployments

### PR DESCRIPTION
**Pull request by AlexBHarley**
### Description

Small fix to the CLI to enable HypFiatCollateral warp route deployments.

My understanding is that all the collateral Hyp tokens require the `collateralAddressOrDenom` property in their warp route config.

You can test this [here](https://github.com/hyperlane-xyz/hyperlane-warp-ui-template/pull/169), just ping me for some of the fake USDC tokens.

### Drive-by changes

N/A

### Related issues

- Fixes [#3778](https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3778)
- Fixes [#3777](https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3777)

### Backward compatibility

Yes

### Testing

CLI and warp UI

### CODE REVIEW

File: typescript/cli/src/deploy/warp.ts [hunk 1]
```diff
@@ -162,7 +162,11 @@ async function getWarpCoreConfig(
     }
 
     const collateralAddressOrDenom =
-      config.type === TokenType.collateral ? config.token : undefined;
+      // todo: should this match typescript/cli/src/config/warp.ts line 136?
+      config.type === TokenType.collateral ||
+      config.type == TokenType.collateralFiat
```
Comment (jmrossy) [inline typescript/cli/src/deploy/warp.ts:167]: Looks like @ltyu added a list recently that would be helpful here: https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3838/files#diff-a1429ad1c3e9d3fe43d28f08333e95f6c79176b6d36873e42e455e5e806826aaR16

Comment (yorhodes) [reply typescript/cli/src/deploy/warp.ts:168]: TODO is fine
will followup when we are merging CLI 2.0, hopefully in a couple days

**Review by jmrossy — COMMENTED** [review]
LGTM but please check with @cmcewen whether this should merge into main or cli2.0

**Comment by ltyu** [comment]
closing this as a dupe to https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4133

**Pull request closed without merging**